unstable unit test TestDiscoveryRunner

ref: https://github.com/hanfei1991/microcosm/runs/6115236387?check_suite_focus=true

```go
{"level":"warn","ts":"2022-04-21T15:55:02.004Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000d2c000/127.0.0.1:34519","attempt":0,"error":"rpc error: code = Canceled desc = context canceled"}
{"level":"warn","ts":"2022-04-21T15:55:11.305Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.305Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.305Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
--- FAIL: TestDiscoveryRunner (9.99s)
    runner_test.go:103: 
        	Error Trace:	runner_test.go:103
        	Error:      	not enough peer update received
        	Test:       	TestDiscoveryRunner
        	Messages:   	received: 18, expected: 20
{"level":"warn","ts":"2022-04-21T15:55:11.306Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.306Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.306Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.308Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.308Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.308Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.308Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.309Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.309Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
FAIL
coverage: 85.4% of statements
{"level":"warn","ts":"2022-04-21T15:55:11.310Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.310Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.306Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.306Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
{"level":"warn","ts":"2022-04-21T15:55:11.306Z","logger":"etcd-client","caller":"v3@v3.5.2/retry_interceptor.go:62","msg":"retrying of unary invoker failed","target":"etcd-endpoints://0xc000c64700/127.0.0.1:44451","attempt":0,"error":"rpc error: code = Unavailable desc = error reading from server: EOF"}
FAIL	github.com/hanfei1991/microcosm/pkg/srvdiscovery	10.159s
```

--- a/pkg/srvdiscovery/runner.go
+++ b/pkg/srvdiscovery/runner.go
@@ -11,6 +11,10 @@ import (
 
 // DiscoveryRunner defines an interface to run Discovery service
 type DiscoveryRunner interface {
+	// ResetDiscovery creates a new discovery service, close the old discovery
+	// watcher and creates a new watcher.
+	// if resetSession is true, the session of discovery runner will be recreated
+	// and returned.
 	ResetDiscovery(ctx context.Context, resetSession bool) (Session, error)
 	GetWatcher() <-chan WatchResp
 	// returns current snapshot, DiscoveryRunner maintains this as snapshot plus
@@ -18,6 +22,7 @@ type DiscoveryRunner interface {
 	// consume more memory, but since it contains node address information only,
 	// the memory consumption is acceptable.
 	GetSnapshot() Snapshot
+	// ApplyWatchResult applies changed ServiceResource to the snapshot of runner
 	ApplyWatchResult(WatchResp)
 }
 

--- a/pkg/srvdiscovery/runner_test.go
+++ b/pkg/srvdiscovery/runner_test.go
@@ -30,14 +30,14 @@ func TestDiscoveryRunner(t *testing.T) {
 		require.Nil(t, err)
 		return NewDiscoveryRunnerImpl(
 			client,
-			3,
+			5,
 			time.Millisecond*50,
 			adapter.NodeInfoKeyAdapter.Encode(string(res.ID)),
 			resStr,
 		)
 	}
 
-	runnerN := 20
+	runnerN := 5
 	runners := make([]DiscoveryRunner, 0, runnerN)
 	for i := 0; i < runnerN; i++ {
 		runners = append(runners, newNodeOnline(i))

